libs/langchain/langchain/vectorstores/kdbai.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
```

Comment:
pandas is optional dependency so should be imported within functions that need it